logrotate.c

```diff
@@ -785,8 +789,10 @@ static int compressLogFile(const char *name, const struct logInfo *log, const st
     compressedName = alloca(strlen(name) + strlen(log->compress_ext) + 2);
     sprintf(compressedName, "%s%s", name, log->compress_ext);
 
-    if ((inFile = open(name, O_RDWR | O_NOFOLLOW)) < 0) {
```

Comment:
We need write access in case we want to shred the file after compression.
https://github.com/logrotate/logrotate/blob/0a900b9435522b1314a39ead26ee7cccc91f5674/logrotate.c#L895-L898

So we might only open with `O_RDONLY` in case of `!(log->flags & LOG_FLAG_SHRED)`.

Side note:
Maybe we should add a check in `shred_file()` whether the passed file-descriptor is opened in write mode (via `fcntl(fd, F_GETFL)` and check for `O_WRONLY` or `O_RDWR`)?

Reply:
Good point about the shred scenario.  Not sure about adding the extra check.  The file descriptor is under logrotate's control, isn't it?

Reply:
Do you want me to add the `(log->flags & LOG_FLAG_SHRED)` to select the proper `open()` flags?  I had missed the `shred` option because I do not use it.

Reply:
Maybe something like:
```c
int flags = O_NOFOLLOW;
/* need write access for shredding */
if (log->flags & LOG_FLAG_SHRED) {
    flags |= O_RDWR;
} else {
    flags |= O_RDONLY;
}
```
?